common/vts.proto

```diff
@@ -15,6 +19,122 @@ message CloseArgs {
 }
 
 message CloseResponse {
+    Status status = 1;
+}
+
+message OpenArgs {
+}
+
+message OpenResponse {
+    Status status = 1;
+}
+
+enum TaType {
+    TA_UNSPECIFIED = 0;
+    TA_RAWPUBLICKEY = 1;
+    // Refer https://github.com/veraison/veraison/issues/76, to introduce certs here!
+    TA_INVALID = 255;
+}
+
+message EndorsementID {
+    AttestationFormat type = 1;
+    google.protobuf.Struct parts = 2;
+}
+
+message Evidence {
+    google.protobuf.Struct value = 1;
+}
+
+message GetEndorsementsRequest{
+    EndorsementID id = 1;
+    Evidence evidence = 2;
+}
+
+message Status {
+    bool result = 1;
+    string error_detail = 2;
+}
+
+message GetEndorsementsResponse {
+    Status status = 1;
+    string details = 2;
+    TrustVector trust_vector = 3;
+    google.protobuf.Struct endorsements = 4;
+}
+
+
+message SwComponentID {
+    AttestationFormat type = 1;
+    google.protobuf.Struct parts = 2;
+}
+
+message SwComponent {
+    SwComponentID id = 1;
+    google.protobuf.Struct attributes = 2;
+}
+
+message AddSwComponentsRequest {
+    repeated SwComponent info = 1;
+}
+
+message AddSwComponentsResponse {
+    Status status = 1;
+}
+
+message GetSwComponentRequest {
+    SwComponentID id = 1;
+}
+
+message GetSwComponentResponse {
+    Status status = 1;
+    google.protobuf.Struct attributes = 2;
+}
+
+message TrustAnchID {
+    AttestationFormat type = 1;
+    google.protobuf.Struct parts = 2;
+}
+
+message TrustAnchorValue {
+    TaType type = 1;
+    google.protobuf.Struct value = 2;
+}
+
+message TrustAnchor {
+    TrustAnchID id = 1;
+    TrustAnchorValue value = 2;
+}
+
+message AddTrustAnchorRequest {
+    TrustAnchor trust_anchor = 1;
+}
+
+message AddTrustAnchorResponse {
+    Status status = 1;
+}
+
+message GetTrustAnchorRequest {
+    TrustAnchID ta_id = 1;
+}
+
+message GetTrustAnchorResponse {
+    Status status = 1;
+    TrustAnchorValue value = 2;
+}
+
+// Provisioner is the Service end point that is used to store
+// and retrieve Software Components and Trust Anchors to and from
+// the Endorsement Store
+service Provisioner {
```

Comment:
@setrofim : Yes, we have discussed this briefly! Store will be removed in the next phase and we will have a Logical VTS Endpoint, abstraction which takes on a Verifier Service, Provisioner Service and Common Service. Will be done in net phase!

Reply:
What was the rationale for separating the Verification and Provisioning services (and thus necessitating a Common service, thus complicating the use of VTS). Why not just have `VTS` service that supports the verification and provisioning APIs (plus whatever other interfaces we might need in the future)?

(IIRC, our original discussion that resulted in separating the provisioning and verification interfaces into separate entities was when the provisioner and the verifier were talking to the endorsemetn store directily; now that we have the VTS that mediates access to it, I'm not sure the old arguments still hold).

Reply:
It seems like we adding actual complexity (having to deal with multiple interfaces -- at least two: common, and provisioner/verifier) for a hypothetical benefit (being able to split VTS across multiple nodes). You're also making the assumption that if we were going to split VTS, it would be along provisioner/verifier lines. I don't see why that would necessarily be the case. 

I think it might be better to err on the side of simplicity. If/when it turns out we want to split VTS services, we can always add the necessary interfaces later -- note that since we taking about interfaces here, it should be fairly straight forward. We can always maintain backwards compatibility by providing a "shim" VTS node implementation that forward requests to individual service nodes.

Reply:
Another (more concrete) reason not to merge the services is that the client-side (i.e., the provisioning and verification frontends) would otherwise need to provide implementations for all the VTS RPCs, when all they need is to interface only the relevant subset (e.g., in the case of provisioning implementing the `common.ProvisionerClient` interface).

Reply:
We only need one client side implementation (at least, for a given channel). Provisioner and Verifier will then both use  the VTS client. There is no reason either the Provisioner or the Verifier need to worry about RPC.

Reply:
The problem I see with this approach is that a VTSClient instantiated on the provisioning pipeline will be able to request verification-specific services.  This looks problematic least-privilege-wise.


But one thing I am not fully grasping (which may be worth discussing IRL) is the added complexity you mention.  For example, these are the [interface](https://github.com/veraison/veraison/blob/provisioning-part2/provisioning/storeclient/istoreclient.go) and the [implementation](https://github.com/veraison/veraison/blob/provisioning-part2/provisioning/storeclient/storeclient_grpc.go) of the VTS client on the provisioning side (they currently sit in the `provisioning-part2` branch): they don't look especially complex to me.

Reply:
This will be resolved during Phase-2 of the Refactor.

Reply:
Having read the discussion more deeply and thinking a bit more, I think, I am aligning my thought process with @setrofim.
Simple single VTS Interface to VTS Clients, with an Access Layer to be built on top of it, giving specific API Access of VTS to specific clients can lead to a simple solution. Looking forward to discuss this aspect during the meeting!